README.md

```diff
@@ -198,6 +198,15 @@ A `TraceApi` object is guaranteed to be returned by both of these calls, even if
 
 A fully detailed overview of the `TraceApi` object is available [here](doc/trace-api.md).
 
+## How does automatic tracing work?
+
+Node.js provides concurrency through as asynchronous execution model. To be able to keep track of the current request across asynchronous execution boundaries we use the [async-listener][] module. It modifies the core async APIs in Node.js, and Promises, to save and restore the 'execution context' across the async bounaries. In addition, we instrument frameworks (like express, hapi, etc. if loaded) to create request contexts for the execution of the incoming requests. Similarly, we instrument RPC libraries (like mysql, redis, outbound http, etc.) to keep track of the latency of these operations and to correlate them to the incoming request.
+
+This `async-listener` based mechanism works great in most cases, but it does have some limitations that can prevent us from being able to properly track execution context:
+
+* It is possible to use JavaScript code that does its own queuing of callback functions – simulating async hops. For example, one may write a http request buffering library that queues requests and then performs them in a batch in one shot. In such a case, when all the callbacks fire, they seem to be executing on behalf of the wrong incoming context. This problem is called the pooling problem or the [user-space queuing problem][queueing-problem], and is a fundamental limitation of JavaScript. If your application uses such code, you will notice that certain portions of your outbound RPCs do not get traced. In such cases we try to work around the problem through monkey patching, or work with the library authors to fix the code to properly propagate context. However, finding problematic code is not always trivial.
+* Presently, it is not possible for `async-listener` to keep track of async transitions in ES7 async/await functions that are available with Node 7+. If your application uses async functions, we will not be properly track RPCs. We do expect to be able to track async/await functions in the future once we switch to the new [async-hooks][] API in Node 8+.
```

Comment:
Should we add a brief note earlier in the README to indicate that tracing requests across async/await functions won't work in Node 7?